pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -513,31 +513,49 @@ type DNSSDConfig struct {
 // See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#ec2_sd_config
```

Comment:
Can you add a comment here explaining that the EC2 service discovery requires AWS API keys or role ARN to authenticate and that basic authentication, authorization and OAuth2 fields are not present on purpose?

Reply:
Sure!